packages/server/src/sdk/app/oauth2/crud.ts

```diff
@@ -48,3 +48,46 @@ export async function get(id: string): Promise<OAuth2Config | undefined> {
   const doc = await getDocument()
   return doc?.configs?.[id]
 }
+
+export async function update(config: OAuth2Config): Promise<OAuth2Config> {
+  const db = context.getAppDB()
+  const doc: OAuth2Configs = (await getDocument(db)) ?? {
+    _id: DocumentType.OAUTH2_CONFIG,
+    configs: {},
+  }
+
+  if (!doc.configs[config.id]) {
+    throw new HTTPError(`OAuth2 config with id '${config.id}' not found.`, 404)
+  }
+
+  if (
+    Object.values(doc.configs).find(
+      c => c.name === config.name && c.id !== config.id
+    )
+  ) {
+    throw new HTTPError("Name is not available.", 400)
```

Comment:
```suggestion
    throw new HTTPError(`OAuth config with name "${config.name}" not found`, 400)
```

Reply:
No, this is meant to be like this. It checks that the name is not already taken for another config, avoiding duplicates

Reply:
Sorry, not "not found" but "already taken." Essentially if "name is not available" were shown to the user I'm not sure they'd know exactly what went wrong. Being more explicit that the chosen name is already taken would be good.